src/renderer/components/dialog-add-version.tsx

```diff
@@ -168,30 +190,44 @@ export class AddVersionDialog extends React.Component<
   }
 
   private renderPath(): JSX.Element | null {
-    const { folderPath, isValidElectron } = this.state;
+    const { folderPath } = this.state;
 
     if (!folderPath) return null;
-
-    const info = isValidElectron
-      ? `We found an ${getElectronNameForPlatform()} in this folder.`
-      : `We did not find a ${getElectronNameForPlatform()} in this folder...`;
-
     return (
       <Callout>
-        {info}
+        {this.renderDialog()}
         {this.renderVersionInput()}
       </Callout>
     );
   }
 
+  private renderDialog(): string {
+    const { existingLocalVersion, isValidElectron } = this.state;
+
+    if (isValidElectron && existingLocalVersion) {
+      return `This folder path has already been added as version "${existingLocalVersion.version}". Would you like to set this path as a new version? \n \n`;
+    } else if (isValidElectron) {
+      return `We found an ${getElectronNameForPlatform()} in this folder.`;
+    }
+
+    return `We did not find a ${getElectronNameForPlatform()} in this folder...`;
+  }
+
   private renderVersionInput(): JSX.Element | null {
-    const { isValidElectron, isValidVersion, version } = this.state;
+    const {
+      existingLocalVersion,
+      isValidElectron,
+      isValidVersion,
+      version,
+    } = this.state;
+    const shouldRemovePreviousVersion = isValidElectron && existingLocalVersion;
```

Comment:
I don't think checking for `isValidElectron` is necessary here because if this variable is false, we're already returning `null` right below.